python/pyarrow/_cuda.pyx

```diff
@@ -159,26 +193,29 @@ cdef class Context:
         return pyarrow_wrap_cudabuffer(cudabuf)
 
     def foreign_buffer(self, address, size):
-        """Create device buffer from device address and size as a view.
+        """Create device buffer from address and size as a view.
 
         The caller is responsible for allocating and freeing the
-        memory as well as ensuring that the memory belongs to the
-        CUDA context that this Context instance holds.
+        memory.
 
         Parameters
         ----------
         address : int
-          Specify the starting address of the buffer.
+          Specify the starting address of the buffer. The address must
+          be accessible from device after mapping it with
+          `get_device_address` method.
```

Comment:
Shouldn't you update the tests for `foreign_buffer`?

Reply:
No, because there are none and the change of documentation has little effect when dealing with device memory addresses. The `foreign_buffer` method is tested implicitly by other methods.